cdc/server.go

```diff
@@ -177,6 +165,10 @@ func (s *Server) Run(ctx context.Context) error {
 		return err
 	}
 
+	// Check cluster version and wait if it's incompatible.
+	// We start status server first to not block tiup cluster upgrading.
+	checkAndWaitClusterVersion(ctx, s.pdClient, s.pdEndpoints, conf.Security)
+
 	return s.run(ctx)
```

Comment:
Currently `/status` API is available after starts server starts, however https://github.com/pingcap/ticdc/issues/2691 is considering change the behavior, which means the `/status` API will be available after capture info is persisted to etcd.